data/service/data-service.js

```diff
@@ -1283,7 +1289,20 @@ exports.DataService = Montage.specialize(/** @lends DataService.prototype */ {
                 useDelegate = isHandler && typeof this.fetchRawObjectProperty === "function",
                 delegateFunction = !useDelegate && isHandler && this._delegateFunctionForPropertyName(propertyName),
                 propertyDescriptor = !useDelegate && !delegateFunction && isHandler && this._propertyDescriptorForObjectAndName(object, propertyName),
-                childService = !isHandler && this._getChildServiceForObject(object);
+                childService = !isHandler && this._getChildServiceForObject(object),
+                debug = exports.DataService.debugProperties.has(propertyName),
+                trace = debug || exports.DataService.traceProperties.has(propertyName);
+
+
+            // Check if property is included in the traceProperties or debugProperties 
+            // collections. Intended for debugging
+            if (trace) {
+                console.log("DataService.fetchObjectProperty", object, propertyName);
+            }
+            
+            if (debug) {
```

Comment:
Event with a debug conditional, we can't have a debugger; statement in production code as it can derail runtime optimizations

Reply:
@marchant 

As a general rule I agree. I saw the pattern used in MontageReviver ([line 351](https://github.com/montagejs/montage/blob/master/core/serialization/deserializer/montage-reviver.js#L351)) and realized it would be useful if applied here. In this case, I found myself regularly writing `if (propertyName == 'aProperty')` in data-service and data-mapping. This change allows debugging a property fetch without modifying any montage code. 

In the end, I agree it is an anti-pattern, but the functionality of the pattern is useful and I do not want to lose it if possible. Do you have any ideas on how it might be accomplished in another way?